File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hashmap;
        vector<vector<string>> ans;

        for(string s: strs){
            string sorted_str = s;
            sort(sorted_str.begin(), sorted_str.end());
            hashmap[sorted_str].push_back(s);

        }
        for(auto& key_value: hashmap){
            vector<string> string_array = key_value.second;
            ans.push_back(string_array);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if strs is None:
            return [[]]

        ans = []
        hashmap = {}
        for string in strs:
            sorted_str = "".join(sorted(string))
            if sorted_str in hashmap:
                hashmap[sorted_str].append(string)
            else:
                hashmap[sorted_str] = [string]

        for anagram, words in hashmap.items():
            ans.append(words)
            
        return ans